### PR TITLE
Fix puspec.yaml path when analysis_option.yaml is not at root

### PR DESCRIPTION
**Pull request by olexale**
Thanks for this package! It makes the process of creating custom rules sooo much simpler, I love it! 💙

The package, though, does not work for cases when `analysis_options.yaml` is placed not near the `pubspec.yaml` file. This is the case for packages with custom linter settings where `analysis_options` is located somewhere inside `lib` folder. With this PR, I propose to introduce additional logic - when `pubspec` is not found, we may iterate through path components and try to find `pubspec` - it must be somewhere, right?

I've checked the package locally with the proposed changes, and it seems to work. Unfortunately, I was not able to write tests for these changes because almost all tests from the `main` branch are failing for me with
```ruby
  FileSystemException: Cannot open file, path = '/Users/ole/dev/pets/dart_custom_lint/packages/example_lint/.dart_tool/package_config.json' (OS Error: No such file or directory, errno = 2)
  dart:io                            _File.readAsStringSync
  test/peer_project_meta.dart 43:40  new PeerProjectMeta.fromDirectory
  test/peer_project_meta.dart 56:42  PeerProjectMeta.current
  test/peer_project_meta.dart        PeerProjectMeta.current
  test/create_project.dart 36:29     createPlugin
  test/server_test.dart 169:20       main.<fn>
  test/server_test.dart 168:51       main.<fn>
```
Is there a missing package?

Anyway, happy to add tests if you agree that this functionality is useful.

### CODE REVIEW

File: packages/custom_lint/lib/src/analyzer_plugin/plugin_link.dart [hunk 1]
```diff
@@ -322,17 +322,37 @@ final allPluginLinksProvider =
...
   );
 }
 
+File _loadRawPubspecAt(String packagePath) {
+  final pubspecFile = File(p.join(packagePath, 'pubspec.yaml'));
+  if (pubspecFile.existsSync()) {
+    return pubspecFile;
+  }
+
+  final pathComponents = p.split(packagePath);
```
Comment (rrousselGit) [inline packages/custom_lint/lib/src/analyzer_plugin/plugin_link.dart:339]: Instead of path manipulation, we can use `File/Directory.parent`

**Comment by rrousselGit** [comment]
Hello!
Sorry for the delay.

This needs some test. Fancy doing it?

**Comment by rrousselGit** [comment]
Hello! Do you still plan on working on this?

**Comment by olexale** [comment]
Hey! Oh, I see that now tests are working on `main`. Sure, I'll update the PR soon.

**Pull request closed without merging**